src/Symfony/Component/Workflow/Command/WorkflowDumpCommand.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Command;
+
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Completion\CompletionInput;
+use Symfony\Component\Console\Completion\CompletionSuggestions;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+use Symfony\Component\Workflow\Debug\TraceableWorkflow;
+use Symfony\Component\Workflow\Dumper\GraphvizDumper;
+use Symfony\Component\Workflow\Dumper\MermaidDumper;
+use Symfony\Component\Workflow\Dumper\PlantUmlDumper;
+use Symfony\Component\Workflow\Dumper\StateMachineGraphvizDumper;
+use Symfony\Component\Workflow\Marking;
+use Symfony\Component\Workflow\StateMachine;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ *
+ * @final
+ */
+#[AsCommand(name: 'workflow:dump', description: 'Dump a workflow')]
+class WorkflowDumpCommand extends Command
+{
+    private const DUMP_FORMAT_OPTIONS = [
+        'puml',
+        'mermaid',
+        'dot',
+    ];
+
+    public function __construct(
+        private ServiceLocator $workflows,
```

Comment:
instead of hardcoding ServiceLocator from the DI component, this should use `Symfony\Contracts\Service\ServiceProviderInterface` as type, making it depend only on the contracts package